datalad/support/gitrepo.py

```diff
@@ -1741,14 +1741,28 @@ def get_hexsha(self, commitish=None, short=False):
 
         Returns
         -------
-        str or, if there are not commits yet, None.
+        str or, if no commitish was given and there are no commits yet, None.
+
+        Raises
+        ------
+        ValueError
+          If a commitish was given, but no corresponding commit could be
+          determined.
         """
-        stdout = self.format_commit("%{}".format('h' if short else 'H'),
-                                    commitish)
-        if stdout is not None:
-            stdout = stdout.splitlines()
-            assert(len(stdout) == 1)
-            return stdout[0]
+        # use --quiet because the 'Needed a single revision' error message
+        # that is the result of running this in a repo with no commits
+        # isn't useful to report
+        cmd = ['rev-parse', '--verify', '{}^{{commit}}'.format(
```

Comment:
The comment above mentions `--quiet`, but it's not passed here.

Reply:
Arrgh, thx, wil fix.

Reply:
Force-pushed an update